pkgs/by-name/li/linux-wallpaperengine/package.nix

```diff
@@ -85,19 +94,28 @@ stdenv.mkDerivation rec {
     libffi
     wayland-scanner
     libXrandr
+    fftw
+    kissfftFloat
   ];
 
   cmakeFlags = [
     "-DCMAKE_BUILD_TYPE=${cef.buildType}"
     "-DCEF_ROOT=${cef}"
-    "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}/app/linux-wallpaperengine"
+    "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}/share/linux-wallpaperengine"
   ];
 
-  preFixup = ''
-    patchelf --set-rpath "${lib.makeLibraryPath buildInputs}:${cef}" $out/app/linux-wallpaperengine/linux-wallpaperengine
-    chmod 755 $out/app/linux-wallpaperengine/linux-wallpaperengine
+  postInstall = ''
+    rm -rf $out/bin $out/lib $out/include
+    chmod 755 $out/share/linux-wallpaperengine/linux-wallpaperengine
     mkdir $out/bin
-    ln -s $out/app/linux-wallpaperengine/linux-wallpaperengine $out/bin/linux-wallpaperengine
+    ln -s $out/share/linux-wallpaperengine/linux-wallpaperengine $out/bin/linux-wallpaperengine
+  '';
+
+  preFixup = ''
+    find $out/share/linux-wallpaperengine -type f -exec file {} \; | grep 'ELF' | cut -d: -f1 | while read -r elf_file; do
+      patchelf --shrink-rpath --allowed-rpath-prefixes "$NIX_STORE" "$elf_file"
+    done
+    patchelf --set-rpath "${lib.makeLibraryPath finalAttrs.buildInputs}:$out/share/linux-wallpaperengine" $out/share/linux-wallpaperengine/linux-wallpaperengine
   '';
```

Comment:
I'm sorry if I don't understand the reason behind this (maybe "hardening the reproducibility") but I'm not sure whether it's actually needed to convolute this. Or did something break?